typing/ctype.ml

```diff
@@ -5093,6 +5082,20 @@ let rec subtype_rec env trace t1 t2 cstrs =
         (trace, t1, t2, !univar_pairs)::cstrs
   end
 
+and subtype_poly env trace tpoly1 tpoly2 cstrs =
+  match tpoly1.poly_params, tpoly2.poly_params with
+  | ([], []) -> subtype_rec env trace tpoly1.poly_body tpoly2.poly_body cstrs
+  | (_ , []) ->
+    let _, u1' = instance_poly ~fixed:false tpoly1 in
+    subtype_rec env  trace u1' tpoly2.poly_body cstrs
+  | (_ , _ ) ->
+    begin try
+      enter_poly env tpoly1 tpoly2
+        (fun t1 t2 -> subtype_rec env trace t1 t2 cstrs)
+    with Escape _ ->
+      (trace, newty (Tpoly tpoly1), newty (Tpoly tpoly2), !univar_pairs)::cstrs
```

Comment:
Do we have a guaranty that the levels are the same as the ones of `subtype_rec`'s `t1` and `t2` here ? Or maybe it doesn't matter ?

Reply:
we indeed change the level here (and more importantly break the sharing). However, my understanding is that it is totally safe because we reach this case if `enter_poly` failed. Thus, when calling `unify` on those values we will one again fail with the call to `enter_poly_for` (`Tpoly` case in `unify3`)

Reply:
Indeed, the `subtype` function is building a list of delayed unification. After any error, the important point is that we record a pair which raises the same error.